src/context/online.rs

```diff
@@ -140,15 +165,73 @@ impl AudioContext {
     /// the time at which the first sample in the buffer is actually processed
     /// by the audio output device.
     #[must_use]
+    #[allow(clippy::missing_panics_doc)]
     pub fn output_latency(&self) -> f64 {
-        self.backend.output_latency()
+        self.backend.lock().unwrap().output_latency()
     }
 
     /// Identifier or the information of the current audio output device.
     ///
     /// The initial value is `None`, which means the default audio output device.
-    pub fn sink_id(&self) -> Option<&str> {
-        self.backend.sink_id()
+    #[allow(clippy::missing_panics_doc)]
+    pub fn sink_id(&self) -> Option<String> {
+        self.backend.lock().unwrap().sink_id().map(str::to_string)
+    }
+
+    /// Update the current audio output device.
+    ///
+    /// This function operates synchronously and might block the current thread. An async version
+    /// is currently not implemented.
+    #[allow(clippy::needless_collect, clippy::missing_panics_doc)]
+    pub fn set_sink_id_sync(&self, sink_id: String) -> Result<(), Box<dyn Error>> {
+        if self.sink_id().as_deref() == Some(&sink_id) {
+            return Ok(()); // sink is already active
+        }
+
+        if !crate::enumerate_devices()
+            .into_iter()
+            .any(|d| d.device_id() == sink_id)
+        {
+            Err("NotFoundError: invalid sinkId")?;
+        }
+
+        let mut backend_guard = self.backend.lock().unwrap();
+
+        // acquire exclusive lock on ctrl msg sender
+        let ctrl_msg_send = self.base.lock_control_msg_sender();
+
+        // flush out the ctrl msg receiver, cache
+        let pending_msgs: Vec<_> = self.render_thread_init.ctrl_msg_recv.try_iter().collect();
+
+        // acquire the audio graph from the current render thread, shutting it down
+        let (graph_send, graph_recv) = crossbeam_channel::bounded(1);
+        let message = ControlMessage::Shutdown { sender: graph_send };
+        ctrl_msg_send.send(message).unwrap();
+        let graph = graph_recv.recv().unwrap();
+
+        // hotswap the backend
+        let options = AudioContextOptions {
+            sample_rate: Some(self.sample_rate()),
```

Comment:
If this sample rate is not available on the new sink, bad things will happen. That is why #183 exists